### PR TITLE
Build: Support overwrite maven repo by gradle properties

### PR DESCRIPTION
**Pull request by pan3793**
cc @RussellSpitzer

### CODE REVIEW

File: deploy.gradle [hunk 1]
```diff
@@ -113,8 +113,8 @@ subprojects {
         // upload to the releases repository using ./gradlew -Prelease publish
         def apacheSnapshotsRepoUrl = 'https://repository.apache.org/content/repositories/snapshots'
         def apacheReleasesRepoUrl = 'https://repository.apache.org/service/local/staging/deploy/maven2'
-        def snapshotsRepoUrl = project.hasProperty('mavenSnapshotsRepo') ? project.property('mavenSnapshotsRepo') : "$apacheSnapshotsRepoUrl"
-        def releasesRepoUrl = project.hasProperty('mavenReleasesRepo') ? project.property('mavenReleasesRepo') : "$apacheReleasesRepoUrl"
+        def snapshotsRepoUrl = project.hasProperty('mavenSnapshotsRepo') ? "$mavenSnapshotsRepo" : "$apacheSnapshotsRepoUrl"
```
Comment (RussellSpitzer) [inline deploy.gradle:116]: I'm a little lost here, it looks like previously we could set this based on the -PmavenSnapshotsRepo and now we are switching it to a local property? Or does -P handle set the local property as well?

Mostly i'm confused why project.property was not sufficient before

Comment (RussellSpitzer) [reply deploy.gradle:116]: Groovy and Gradle still hold a lot of mysteries for me

Comment (RussellSpitzer) [reply deploy.gradle:116]: From my understanding project.property('mavenSnapshotsRepo') picks up properties from either a gradles.properties file or from -P args, I"ll run some tests to see

Comment (pan3793) [reply deploy.gradle:116]: yeap, and -P args should have the high priority.

Comment (pan3793) [reply deploy.gradle:116]: emm... github webpage looks good to me.

Comment (pan3793) [reply deploy.gradle:116]: oh, i see. the inlined conversation diff looks weird, but `File Changed` page looks good.

Comment (RussellSpitzer) [reply deploy.gradle:116]: I'm still lost a bit I think
```groovy
  def snapshotsRepoUrl = project.hasProperty('mavenSnapshotsRepo') ? "$mavenSnapshotsRepo" : "$apacheSnapshotsRepoUrl"
```

should be 
```groovy
  def snapshotsRepoUrl = project.hasProperty('mavenSnapshotsRepo') ? project.property("mavenSnapshotsRepo") : "$apacheSnapshotsRepoUrl"
  ```
  
  When I do a quick test
  
```groovy
print(project.hasProperty("mavenSnapshotsRepo"))
print(project.property("mavenSnapshotsRepo"))
print($mavenSnapshotsRepo)
```

and
```bash
 g -PmavenSnapshotsRepo=one
```

Then I get mavenSnapshotsRepo undefined but the first two work fine.

Comment (pan3793) [reply deploy.gradle:116]: `print("$mavenSnapshotsRepo")` works but `print($mavenSnapshotsRepo)` not.

"$mavenSnapshotsRepo" will fallback to `project.property("mavenSnapshotsRepo")`

Comment (RussellSpitzer) [reply deploy.gradle:116]: So this is some magic in the string interpolation?

Comment (pan3793) [reply deploy.gradle:116]: yes

**Comment by RussellSpitzer** [comment]
@aokolnychyi + @rdblue I haven't done a release so one of you should probably check it too :)

**Review by RussellSpitzer — APPROVED** [review]

**Review by rdblue — APPROVED** [review]
Looks fine to me.

**Pull request merged**